File: packages/tasks-gen/snippets-fixtures/image-to-video/js/fetch/0.fal-ai.js
const image = fs.readFileSync("cat.png");

async function query(data) {
	const response = await fetch(
		"https://router.huggingface.co/fal-ai/<fal-ai alias for Wan-AI/Wan2.2-I2V-A14B>?_subdomain=queue",
		{
			headers: {
				Authorization: `Bearer ${process.env.HF_TOKEN}`,
				"Content-Type": "image/jpeg",
			},
			method: "POST",
			body: {
				"image_url": `data:image/png;base64,${data.image.encode("base64")}`,
				"prompt": data.prompt,
			}
		}
	);
	const result = await response.json();
	return result;
}

query({
	"image": image,
	"prompt": "The cat starts to dance",
}).then((response) => {
    // Use video
});

File: packages/tasks-gen/snippets-fixtures/image-to-video/python/requests/0.fal-ai.py
import os
import base64
import requests

API_URL = "https://router.huggingface.co/fal-ai/<fal-ai alias for Wan-AI/Wan2.2-I2V-A14B>?_subdomain=queue"
headers = {
    "Authorization": f"Bearer {os.environ['HF_TOKEN']}",
}

def query(payload):
    with open(payload["inputs"], "rb") as f:
        img = f.read()
        payload["inputs"] = base64.b64encode(img).decode("utf-8")
    response = requests.post(API_URL, headers=headers, json=payload)
    return response.content

video_bytes = query({
    "inputs": "cat.png",
    "parameters": {
        "prompt": "The cat starts to dance"
    }
})